Microsoft Windows PowerShell 2.1 (#1434)
mulesoft/docs-connectors: windows-powershell/2.1/modules/ROOT/pages/microsoft-powershell-connector-reference.adoc

|===

[[config_windows]]
=== Windows Configuration

==== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| IPAddress a| String |  Host used to initialize the session. |  |x
| UserName a| String |  Username used to initialize the session. The value for this field, can be, for example: `username` or `username@domain`|  |x
| Password a| String |  Password used to authenticate the user. |  |x
| Reconnection a| Reconnection |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy |  |
|===

== Operations

* <<executeCommand>>
* <<executeScriptFile>>
* <<executeCommandExceptionHandlingEnabled>>
* <<executeScriptFileExceptionHandlingEnabled>>

[[executeCommand]]
== Execute Command

`<powershell:execute-command>`

This operation executes the command on the machine.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | |x

> | Command a| String |  The command to Execute on the client machine. |  |x

Reviewer: ```suggestion
| Command a| String |  The command to execute on the client machine. |  |x
```

> | Target Variable a| String |  The name of a variable on which the operation's output is placed. |  |

Reviewer: ```suggestion
| Target Variable a| String |  Name of the variable that stores the operation's output. |  |
```

> | Target Value a| String |  An expression to evaluate against the operation's output. The outcome of that expression is stored in the target variable. |  `#[payload]` |

Reviewer: ```suggestion
| Target Value a| String |  Expression that evaluates the operation's output. The expression outcome is stored in the target variable. |  `#[payload]` |
```

| Target Variable a| String |  The name of a variable on which the operation's output is placed. |  |
| Target Value a| String |  An expression to evaluate against the operation's output. The outcome of that expression is stored in the target variable. | ` #[payload]` |
| Reconnection Strategy a| * reconnect
* reconnect-forever |  A retry strategy in case of connectivity errors. |  |
|===

=== Output

[%header%autowidth.spread]
|===
| Type a| String
|===

=== For Configurations

* config

=== Throws

* POWERSHELL:RETRY_EXHAUSTED
* POWERSHELL:CONNECTIVITY

[[executeCommandExceptionHandlingEnabled]]
== Execute Command (exception handling enabled)

`<powershell:execute-command-exception-handling-enabled>`

This operation executes the command on the machine and throws execution related exceptions, if any.

=== Parameters

[%header%autowidth.spread]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | |x
| Command a| String |  The command to execute on the client machine. |  |x
| Target Variable a| String |  The name of a variable on which the operation's output is placed. |  |

> | Target Value a| String |  An expression to evaluate against the operation's output. The outcome of that expression is stored in the target variable. |  `#[payload]` |

Reviewer: Same as stated previously.
